core/ledger/pvtdatastorage/helper.go

```diff
@@ -181,6 +189,41 @@ func prepareHashedIndexEntries(dataEntires []*dataEntry) ([]*hashedIndexEntry, e
 	return hashedIndexEntries, nil
 }
 
+func preparePurgerMarkerEntries(blkNum uint64, purgeMarkers []*PurgeMarker) ([]*purgeMarkerEntry, []*purgeMarkerCollEntry) {
+	purgeMarkersEntries := []*purgeMarkerEntry{}
+	purgeMarkersCollEntries := []*purgeMarkerCollEntry{}
+
+	for _, m := range purgeMarkers {
+		purgeMarkersEntries = append(purgeMarkersEntries,
+			&purgeMarkerEntry{
+				key: &purgeMarkerKey{
+					ns:         m.Ns,
+					coll:       m.Coll,
+					pvtkeyHash: m.PvtkeyHash,
+				},
+				value: &purgeMarkerVal{
+					blkNum: blkNum,
+					txNum:  m.TxNum,
+				},
+			},
+		)
+
+		purgeMarkersCollEntries = append(purgeMarkersCollEntries,
+			&purgeMarkerCollEntry{
+				key: &purgeMarkerCollKey{
+					ns:   m.Ns,
+					coll: m.Coll,
+				},
+				value: &purgeMarkerVal{
+					blkNum: blkNum,
+					txNum:  m.TxNum,
+				},
+			},
```

Comment:
The RFC mentions the regular PurgeMarker, but not the PurgeMarker collection entry. Could you remind me what the PurgeMarker collection entry will be used for, and update the RFC accordingly?

There could be many purged keys in the same collection, resulting in many duplicates of the PurgeMarker collection entry. I guess when the leveldb batch is updated each occurrence will overwrite the same record and therefore only one record will eventually be written per collection. Is this the design you intended here?

Reply:
This change is not introduced in this PR. The PurgeMarker collection entry was introduced in the PR that filters and deletes in the background. This is used [here](https://github.com/hyperledger/fabric/blob/811a36d8ab82e8a37640cbb521f1db641e5ac321/core/ledger/pvtdatastorage/store.go#L553) to reduce the overheads for the existing deployments until they start using this feature.

Yes, there would be a single entry per collection that we keep overwriting.